sycl/include/CL/sycl/access/access.hpp

```diff
@@ -115,13 +115,15 @@ constexpr bool modeWritesNewData(access::mode m) {
 #define __OPENCL_LOCAL_AS__ __attribute__((opencl_local))
 #define __OPENCL_CONSTANT_AS__ __attribute__((opencl_constant))
 #define __OPENCL_PRIVATE_AS__ __attribute__((opencl_private))
+#define __OPENCL_GENERIC_AS__ __attribute__((opencl_generic))
```

Comment:
What do we need `opencl_generic` attribute for? I suggest avoiding it and use default address space instead.

Reply:
It was used for fallback mechanism when pointer from non-global and non-local AS were casted to global through generic cast.

I have reverted that behavior and raise assert in that case because otherwise behavior is undefined.